### DISCUSSION TITLE
How to mock auto inputs to test `prompt` function

### DISCUSSION THREAD

**Discussion by aliuq** [denoland/deno#15182] [Q&A]
I write a function to wrap `prompt` function, and I trying to write a test case, I don't know how to mock user input, Have any ideas to mock it?

```ts
import { assertEquals } from 'https://deno.land/std@0.147.0/testing/asserts.ts'

Deno.test('Function - wrapPrompt', async () => {
  const res = await wrapPrompt('prompt') // Terminal stoped here
  await Deno.stdout.write(new TextEncoder().encode('y\n'))
  assertEquals(res, 'y')
})
```

**Comment by zachauten** [comment]
You'll want to use the [stub](https://deno.land/manual/testing/mocking#stubbing) function from the testing lib. `prompt` exists on the global object (window), so create the stub like this:
```
import { stub } from "https://deno.land/std@0.148.0/testing/mock.ts";

// Calls to prompt will always return the same string.
const prompt_stub = stub(window, "prompt", (s) => "mocked input value");

// Restore prompt to it's un-stubbed state
prompt_stub.restore();
```

**Reply by aliuq** [reply]
There may also be some internal errors

![image](https://user-images.githubusercontent.com/11707934/179237468-b7092b1b-e655-491f-abec-f8ecc0b2a609.png)

**Reply by zachauten** [reply]
Hard to say what your code is doing without seeing wrapPrompt

**Reply by aliuq** [reply]
wrapPrompt is here

https://github.com/aliuq/deno_module/blob/master/lib/utils.ts#L52

**Reply by zachauten** [reply]
I would try stepping through it in the debugger to see what args `prompt` is called with.
